### PR TITLE
read workdir from git.config as referenced in man 1 git-config

### PR DESCRIPTION
**Pull request by Farom**
The setting in .git/config for GIT_WORK_TREE were ignored.
I tried to implement this as git-config says.

### CODE REVIEW

File: git/repo/base.py [hunk 1]
```diff
@@ -140,7 +140,20 @@ def __init__(self, path=None, odbt=GitCmdObjectDB, search_parent_directories=Fal
...
+                # variable. The value can be an absolute path or relative to the path to the .git
+                # directory, which is either specified by GIT_DIR, or automatically discovered.
+                # If GIT_DIR is specified but none of GIT_WORK_TREE and core.worktree is specified,
+                # the current working directory is regarded as the top level of your working tree.
+                self._working_tree_dir = os.path.dirname(self.git_dir)
+                if os.environ.get('GIT_COMMON_DIR') is None:
+                    gitconf = self.config_reader("repository")
+                    if gitconf.has_option('core', 'worktree'):
+                        self._working_tree_dir = gitconf.get('core', 'worktree')
+                self._working_tree_dir = os.getenv('GIT_WORK_TREE', self._working_tree_dir)
```
Comment (Byron) [inline git/repo/base.py]: This looks like in the end, the old behaviour is always taking precedence, as `_working_tree_dir` is written unconditionally.

**Comment by Byron** [comment]
Thanks a lot for your contribution! It's much appreciated.
I also added a comment about something puzzling, maybe you can have a look.
Thank you

File: git/repo/base.py (hunk 1 above)
Comment (Farom) [reply git/repo/base.py]: I do not think so:

Set _working_tree_dir to git_dir.
if env(GIT_COMMON_DIR) is unset, set  _working_tree_dir to core.worktree, if it exists.
If env(GIT_WORK_TREE) is set, use it or use the value before.

Do you think line 156 would be more pythonic this way:?

```
if os.environ.has_key('GIT_WORK_TREE'):
     self._working_tree_dir = os.getenv('GIT_WORK_TREE')
```

Comment (Byron) [reply git/repo/base.py]: Thank you @ukleinek , thanks @Farom , I think I understand now. Given that I was easily tricked by it, maybe that indicates we should indeed put the code behind an `if` condition. That way, we don't do all the work if, in the end, we are willing to throw it away for the value which is in `GIT_WORK_TREE`.
Once this change is made, I believe we can merge this one.

**Comment by Byron** [comment]
One change request was made, after which this PR can be merged. See the resolved conversation in `git/repo/base`.

**Comment by Byron** [comment]
Thanks a lot for your contribution thus far !
I am closing this PR due to prolonged inactivity. Please feel free to reopen it in case it can proceed.

**Pull request closed without merging**